lib/Bootstrap.php

```diff
@@ -226,23 +228,24 @@ public static function defineConstants()
 
     public static function autoload()
     {
-        $loader = \Pimcore::getAutoloader();
-
         // tell the autoloader where to find Pimcore's generated class stubs
         // this is primarily necessary for tests and custom class directories, which are not covered in composer.json
-        $loader->addPsr4('Pimcore\\Model\\DataObject\\', PIMCORE_CLASS_DIRECTORY . '/DataObject');
+        $dataObjectClassLoader = new ClassLoader();
+        $dataObjectClassLoader->addPsr4('Pimcore\\Model\\DataObject\\', PIMCORE_CLASS_DIRECTORY . '/DataObject');
+        $dataObjectClassLoader->register(true);
 
         // compatibility autoloader for the \Pimcore\Model\Object\* namespace (seems to work with PHP 7.2 as well, tested with 7.2.3)
-        $dataObjectCompatibilityLoader = new \Pimcore\Loader\Autoloader\DataObjectCompatibility($loader);
-        $dataObjectCompatibilityLoader->register(true);
+        $dataObjectCompatibilityLoader = new DataObjectCompatibility();
+        $dataObjectCompatibilityLoader->register();
 
         // legacy mapping loader creates aliases for renamed classes
-        $legacyMappingLoader = new \Pimcore\Loader\Autoloader\AliasMapper($loader);
+        $legacyMappingLoader = new AliasMapper();
         $legacyMappingLoader->register(true);
 
         // the following code is out of `app/autoload.php`
         // see also: https://github.com/symfony/symfony-demo/blob/master/app/autoload.php
-        AnnotationRegistry::registerLoader([$loader, 'loadClass']);
+        // Don't think this is necessary anymore
```

Comment:
Let's just kick this code if it's not necessary anymore 😉

Reply:
done